src/backend/distributed/sql/udfs/citus_job_status/latest.sql

```diff
@@ -0,0 +1,86 @@
+CREATE OR REPLACE FUNCTION pg_catalog.citus_job_status (
+    job_id bigint DEFAULT NULL,
+    raw boolean DEFAULT FALSE
+)
+    RETURNS TABLE (
+            job_id bigint,
+            state pg_catalog.citus_job_status,
+            job_type name,
+            description text,
+            started_at timestamptz,
+            finished_at timestamptz,
+            details jsonb
+    )
+    LANGUAGE SQL
+    STRICT
+    AS $fn$
+    WITH rp AS MATERIALIZED (
+        SELECT * FROM get_rebalance_progress()
```

Comment:
This needs a GROUP BY, now it outputs one row per shard in the shard group that is being moved, which is then turned into one entry in the task array.

The following worked well in my testing:
```
        SELECT
            sessionid,
            sum(source_shard_size) as source_shard_size,
            sum(target_shard_size) as target_shard_size,
            any_value(status) as status,
            any_value(sourcename) as sourcename,
            any_value(sourceport) as sourceport,
            any_value(targetname) as targetname,
            any_value(targetport) as targetport,
            max(source_lsn) as source_lsn,
            min(target_lsn) as target_lsn
        FROM get_rebalance_progress()
        GROUP BY sessionid
```

Reply:
I tried to break this query but I was not successful.

I had only one concern: Does the shards in a shard group always have matching states in get_rebalance_progress output. Apparently they do. Even when a shard in a shard group contains significantly smaller items, the status is always the same for both shards. You can check phase attributes that store status of get_rebalance_progress output below.

```
 {                                                                                                                      +
     "tasks": [                                                                                                         +
         {                                                                                                              +
             "LSN": {                                                                                                   +
                 "source": "0/20E14B60",                                                                                +
                 "target": "0/0"                                                                                        +
             },                                                                                                         +
             "size": {                                                                                                  +
                 "source": "14 MB",                                                                                     +
                 "target": "14 MB"                                                                                      +
             },                                                                                                         +
             "hosts": {                                                                                                 +
                 "source": "localhost:9701",                                                                            +
                 "target": "localhost:9702"                                                                             +
             },                                                                                                         +
             "phase": "Copying Data",                                                                                   +
             "state": "running",                                                                                        +
             "command": "SELECT pg_catalog.citus_move_shard_placement(102008,'localhost',9701,'localhost',9702,'auto')",+
             "message": "",                                                                                             +
             "retried": 0,                                                                                              +
             "task_id": 5                                                                                               +
         },                                                                                                             +
         {                                                                                                              +
             "LSN": {                                                                                                   +
                 "source": "0/20E14B60",                                                                                +
                 "target": "0/0"                                                                                        +
             },                                                                                                         +
             "size": {                                                                                                  +
                 "source": "346 MB",                                                                                    +
                 "target": "95 MB"                                                                                      +
             },                                                                                                         +
             "hosts": {                                                                                                 +
                 "source": "localhost:9701",                                                                            +
                 "target": "localhost:9702"                                                                             +
             },                                                                                                         +
             "phase": "Copying Data",                                                                                   +
             "state": "running",                                                                                        +
             "command": "SELECT pg_catalog.citus_move_shard_placement(102008,'localhost',9701,'localhost',9702,'auto')",+
             "message": "",                                                                                             +
             "retried": 0,                                                                                              +
             "task_id": 5                                                                                               +
         }                                                                                                              +
     ],                                                                                                                 +
     "task_state_counts": {                                                                                             +
         "blocked": 3,                                                                                                  +
         "running": 1                                                                                                   +
     }                                                                                                                  +
 }
```